File: cpp/selectionSort.cpp
#include <bits/stdc++.h>

void selectionSort(int arr[], int size) {
    for(int i=0;i<size;i++) {
        int index = i;
        for(int j=i+1;j<size;j++) {
            if(arr[j]<arr[index]) {
                index = j;
            }
        }
        int smallerNum = arr[index];
        arr[index] = arr[i];
        arr[i] = smallerNum;
    }
}

int main() {
    int arr[] = {27, 6, 7, 10, 19, 22, 40, 25};  // This is a C array.
    // sizeof(arr) will give you the number of bytes in a C array.
    // So sizeof(arr) divided by sizeof(arr[0]) (or sizeof(int)) will give you the number of elements in the array.
    // We have to pass the size of the array to the selectionSort() as it's not possible to get the size of the array inside that function.
    int arr_size = sizeof(arr)/sizeof(arr[0]);
    selectionSort(arr, arr_size);
    std::cout<< "Sorted array using Selection sort: " << std::endl;
    for(int i=0;i<arr_size;i++) {
        std::cout << arr[i] << " " << std::endl;
    }
    return 0;
}

File: java/sort/SelectionSort.java
package sort;

public class SelectionSort {
    public static void selectionSort(int arr[]) {
        for(int i=0;i<arr.length-1;i++) {
            int index = i;
            for(int j=i+1;j<arr.length;j++) {
                if(arr[j]<arr[index]) {
                    index = j;
                }
            }
            int smallestNum = arr[index];
            arr[index] = arr[i];
            arr[i] = smallestNum;
        }
    }

    public static void main(String a[]) {
        int arr[] = {27, 6, 7, 10, 19, 22, 40, 25};
        selectionSort(arr);
        System.out.println("Sorted array using Selection sort: ");
        for(int i = 0;i < arr.length;i++) {
            System.out.println(arr[i]+" ");
        }
    }
}
